### PR TITLE
https://github.com/certbot/certbot/issues/6466

### PR DESCRIPTION
**Pull request by b3n4kh**
Test for existing 301 redirect in vhost as described in https://github.com/certbot/certbot/issues/6466.

Fixed two comments since it isnt any rewrite directive (anymore?) but a return 301

### CODE REVIEW

**Comment by bmw** [comment]
On IRC @b3n4kh asked:
> The test would be pretty much the same as the already exsiting test_split_for_redirect() from configurator_test.py but I would either have to "add_server_directives" to the generated http block or write my own "sites-enabled" config
> Or is an test for an edge case like this nonsense? Aany feedback would be really appreciated

I suggested they open a PR and let the reviewer decide if a test is needed.

@ohemorange or @sydneyli, are you able to review this small nginx PR?

**Comment by ohemorange** [comment]
Thanks for the PR @b3n4kh! My personal opinion is that it's not worth it since it doesn't break existing configurations, but maybe @sydneyli thinks otherwise.

File: certbot-nginx/certbot_nginx/_internal/configurator.py [hunk 1]
```diff
@@ -719,7 +719,11 @@ def enhance(self, domain, enhancement, options=None):
 
     def _has_certbot_redirect(self, vhost, domain):
         test_redirect_block = _test_block_from_block(_redirect_block_for_domain(domain))
-        return vhost.contains_list(test_redirect_block)
+        redirect_block_for_server = [['\n    ', 'return', '301', 'https://$host$request_uri']]
```
Comment (sydneyli) [inline certbot-nginx/certbot_nginx/_internal/configurator.py]: This seems specific to the config in #6466. We might want something more generic that can detect the same directive with any tabbing.

And since this is a regular list, I think you need the whitespace information between tokens. Might be able to reuse some existing code in `_redirect_block_for_domain(domain)` here!

Comment (b3n4kh) [reply certbot-nginx/certbot_nginx/_internal/configurator.py]: The tabbing is actually obselete now since I use `nginxparser.UnspacedList`.

I also added a unit Test if `_has_certbot_redirect` returns True if a custom block exists.
